### DISCUSSION TITLE
webgl build and C# cli

### DISCUSSION THREAD

**Discussion by boris-741** [kniEngine/kni#687] [BlazorGL (BlazorWasm/WebGL)]
Are there any instructions on how I can install a template for webassembler project Kni from the command line?

I installed mg:
dotnet new --install MonoGame.Templates.CSharp
dotnet new mgdesktopgl -o MyGame
project create build and start successfully

What should I do now to submit your template for a web assembler project?
Thank you!

**Comment by nkast** [comment]
Install the [KniSdkSetup3.10.9001.exe](https://github.com/kniEngine/kni/releases/download/v3.10.9001/KniSdkSetup3.10.9001.exe) from https://github.com/kniEngine/kni/releases/tag/v3.10.9001

Then open VS2022 and create a new project using either 'KNI Web browser Platform Project (.net6)' 
or 'KNI Multiplatform Project (.net6)' which creates WindowsDX,DesktopGL, Android, and BlazorGL projects in a single solution with shared content.

If you don't see the templates the first time, run '"C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\devenv" /updateconfiguration' in the command line.
It take a while for VS to see new templates.

**Reply by Skyedra** [reply]
I have the same question (how to get started making a webgl and/or multiplatform build), but for linux / dotnet cli (vs code).  

Is there a template up on nuget?  ex using the dotnet syntax in the OP's post

**Reply by nkast** [reply]
Currently no. I've started a branch but I'll not work on it until after the next release.
Additionally, the pipeline builder would not work on linux, because it doesn't have the native libraries.
The frameworks is targeting primarily Windows as the authoring platform.

**Reply by Skyedra** [reply]
Understood; sounds like I won't be able to use the lib unfortunately, so I'd like to respectfully suggest linux support in future.  Monogame works well here, but I was hoping for a route to export HTML5 which is why kni looked quite exciting.  Besides devs who daily drive linux, it's also a popular choice for build automation so I think linux support would be a good addition in future, time permitting.
